pkg/scheduler/framework/v1alpha1/framework_test.go

```diff
@@ -20,160 +20,105 @@ import (
+	return injectNormalizeRes(pl.inj, scores)
 }
 
-func (pl *TestScorePlugin1) Score(pc *PluginContext, p *v1.Pod, nodeName string) (int, *Status) {
-	// Score is currently not used in the tests so just return some dummy value.
-	return 0, nil
+func (pl *ScoreWithNormalizePlugin1) Score(pc *PluginContext, p *v1.Pod, nodeName string) (int, *Status) {
+	return injectScoreRes(pl.inj)
 }
 
-type TestScorePlugin2 struct{}
-
-// NewScorePlugin2 is the factory for NormalizeScore plugin 2.
-func NewScorePlugin2(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
-	return &TestScorePlugin2{}, nil
+type ScoreWithNormalizePlugin2 struct {
+	inj injectedResult
 }
 
-func (pl *TestScorePlugin2) Name() string {
-	return scorePlugin2
+func (pl *ScoreWithNormalizePlugin2) Name() string {
+	return scoreWithNormalizePlugin2
 }
 
-func (pl *TestScorePlugin2) NormalizeScore(pc *PluginContext, pod *v1.Pod, scores NodeScoreList) *Status {
-	// Simply force each node score to 5.
-	for i := range scores {
-		scores[i].Score = 5
-	}
-	return nil
+func (pl *ScoreWithNormalizePlugin2) NormalizeScore(pc *PluginContext, pod *v1.Pod, scores NodeScoreList) *Status {
+	return injectNormalizeRes(pl.inj, scores)
 }
 
-func (pl *TestScorePlugin2) Score(pc *PluginContext, p *v1.Pod, nodeName string) (int, *Status) {
-	// Score is currently not used in the tests so just return some dummy value.
-	return 0, nil
+func (pl *ScoreWithNormalizePlugin2) Score(pc *PluginContext, p *v1.Pod, nodeName string) (int, *Status) {
+	return injectScoreRes(pl.inj)
 }
 
-// TestScorePlugin3 only implements ScorePlugin interface.
-type TestScorePlugin3 struct{}
-
-// NewScorePlugin3 is the factory for Score plugin 3.
-func NewScorePlugin3(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
-	return &TestScorePlugin3{}, nil
+// ScorePlugin1 only implements ScorePlugin interface.
+type ScorePlugin1 struct {
+	inj injectedResult
 }
 
-func (pl *TestScorePlugin3) Name() string {
-	return scorePlugin3
+func (pl *ScorePlugin1) Name() string {
+	return scorePlugin1
 }
 
-func (pl *TestScorePlugin3) Score(pc *PluginContext, p *v1.Pod, nodeName string) (int, *Status) {
-	// Score is currently not used in the tests so just return some dummy value.
-	return 0, nil
+func (pl *ScorePlugin1) Score(pc *PluginContext, p *v1.Pod, nodeName string) (int, *Status) {
+	return injectScoreRes(pl.inj)
 }
 
 // PluginNotImplementingScore doesn't implement the ScorePlugin interface.
 type PluginNotImplementingScore struct{}
 
-// NewPluginNotImplementingScore is the factory for PluginNotImplementingScore.
-func NewPluginNotImplementingScore(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
-	return &PluginNotImplementingScore{}, nil
-}
-
 func (pl *PluginNotImplementingScore) Name() string {
 	return pluginNotImplementingScore
 }
 
-var registry = Registry{
+var constructors = map[string]func(injectedResult) Plugin{
```

Comment:
How about declaring it in the func `toRegistry` that will use it?

Reply:
I decided to pass `constructors` as an argument. This could be used for future tests for other plugins.